### DISCUSSION TITLE
Duplicate definitions on STM32L476xx (OPT_MCU_STM32L4)

### DISCUSSION THREAD

**Discussion by Iktek** [hathach/tinyusb#1436] [Q&A] [answered]
When trying to compile tinyusb on STM32L476xx, I get several duplicate definitions seen below.
When removing the dcd_synopsys.c file the issue is solved. 
It seems that for this particular cpu there are two driver cores active by definition.
Or maybe I put some wrong defines?

Regards Pascal

Libs/tinyusb/src/portable/synopsys/dwc2/dcd_dwc2.c:469: multiple definition of `dcd_init'
Libraries/tinyusb/portable/st/synopsys/dcd_synopsys.o:
Libs/tinyusb/src/portable/st/synopsys/dcd_synopsys.c:463: first defined here
Libraries/tinyusb/portable/synopsys/dwc2/dcd_dwc2.o: In function `tu_edpt_dir':
Libs/tinyusb/src/common/tusb_types.h:496: multiple definition of `dcd_int_enable'
Libraries/tinyusb/portable/st/synopsys/dcd_synopsys.o:Libs/STM32Cube_FW_L4_V1.13.0/Drivers/CMSIS/Include/core_cm4.h:1627: first defined here
Libraries/tinyusb/portable/synopsys/dwc2/dcd_dwc2.o: In function `tu_edpt_dir':
Libs/tinyusb/src/common/tusb_types.h:496: multiple definition of `dcd_int_disable'
Libraries/tinyusb/portable/st/synopsys/dcd_synopsys.o:/Libs/STM32Cube_FW_L4_V1.13.0/Drivers/CMSIS/Include/core_cm4.h:1627: first defined here
Libraries/tinyusb/portable/synopsys/dwc2/dcd_dwc2.o: In function `tu_edpt_dir':
Libs/tinyusb/src/common/tusb_types.h:496: multiple definition of `dcd_set_address'
...

**Comment by hathach** [comment, marked as answer]
dcd_synopsys.c should be removed, it is legacy driver, and is replaced by dcd_dwc2.c . I will remove the dcd_synopsys.c in upcoming commit.

**Reply by umichxuwentao** [reply]
Wow it helps thanks!
